src/virt.c

```diff
@@ -536,9 +543,9 @@ static int lv_read(user_data_t *ud) {
   if ((last_refresh == (time_t)0) ||
       ((interval > 0) && ((last_refresh + interval) <= t))) {
     if (refresh_lists(inst) != 0) {
-      if (conn != NULL)
-        virConnectClose(conn);
-      conn = NULL;
+      if (inst->id == 0) {
```

Comment:
Yeah, that's what I thought. Especially since it got it right a few lines higher.